.github/workflows/podvm_mkosi.yaml

```diff
@@ -97,32 +216,91 @@ jobs:
           remove-docker-images: "true"
 
       - name: Second checkout
+        if: ${{ inputs.arch == 'amd64' }}
         uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: "${{ inputs.git_ref }}"
 
-      - name: Decide on image to use
-        id: binaries-image
+
+      - name: Rebase the code
+        if: github.event_name == 'pull_request_target'
+        working-directory: ./
         run: |
-          if [ -z "${{ github.event.inputs.binaries-image }}" ]; then
-            echo "image=ghcr.io/${{ github.repository }}/podvm/binaries-fedora:${{ github.sha }}" | tee -a "$GITHUB_OUTPUT"
-          else
-            echo "image=${{ github.event.inputs.binaries-image }}" | tee -a "$GITHUB_OUTPUT"
-          fi
+          ./hack/ci-helper.sh rebase-atop-of-the-latest-target-branch
 
       - name: Download binaries and unpack into binaries-tree
         run: |
           crane export \
-            ${{ steps.binaries-image.outputs.image }} \
+            ${{ steps.build_binaries.outputs.image }} \
             podvm-binaries.tar
           mkdir -p podvm-mkosi/resources/binaries-tree
           tar xf podvm-binaries.tar -C podvm-mkosi/resources/binaries-tree
           rm podvm-binaries.tar
 
-      - name: Build image
+      - name: Build mkosi debug image
         working-directory: src/cloud-api-adaptor/podvm-mkosi
-        run: make image-debug
+        run: |
+          if [ "${{inputs.debug}}" = "true" ]; then
+            echo "Making debug image"
+            make image-debug
+          else
+            echo "Making non-debug image"
+            make image
+          fi
+
+      - name: Upload the qcow2 with oras
+        id: publish_oras_qcow2
+        working-directory: src/cloud-api-adaptor/podvm-mkosi
+        run: |
+          mkdir oras
+          cd oras
+          cp ../build/podvm-*.qcow2 .
+          tar cJf podvm.tar.xz podvm-*.qcow2
```

Comment:
are we missing a step? I thought make image was creating a raw image, so we need to convert it to qcow2

Reply:
the `make image` and `make image-debug` targets create the qcow2, mostly as the s390x one already did that, so having them all the same was easier, if you'd prefer that to be separated out let me know.

Reply:
no, that's fine, since raw images are still available.